storage/local/interface.go

```diff
@@ -88,21 +96,9 @@ type SeriesIterator interface {
 	ValueAtOrBeforeTime(model.Time) model.SamplePair
 	// Gets all values contained within a given interval.
 	RangeValues(metric.Interval) []model.SamplePair
```

Comment:
Note how I added error return values to the `Querier` and `Storage` interface methods, since things can go wrong in the real world (not all internal storage failures get plumbed through though).

The `SeriesIterator` methods here could be another candidate for adding error return values. That would enable lazy iterators to communicate failures to lazily load data.